src/utils.ts

```diff
@@ -324,6 +331,62 @@ export function assertNoMissingTypes(schema: GraphQLSchema, missingTypes: Record
   }
 }
 
+export function assertAbstractTypesCanBeDiscriminated(schema: NexusGraphQLSchema, features: NexusFeatures) {
+  // If backing type check is enabled, we can no longer know for sure if a type is properly discriminated
+  // since it could be discriminated via the `__typename` field in resolvers, which we can't check at runtime
+  if (features.abstractTypes?.__typename === true) {
+    return
+  }
+
+  const abstractTypes = Object.values(schema.getTypeMap()).filter(isAbstractType)
+
+  abstractTypes.forEach((type) => {
+    const kind = isInterfaceType(type) ? 'Interface' : 'Union'
+    const resolveTypeImplemented = type.resolveType !== undefined
+    const typesWithoutIsTypeOf = schema.getPossibleTypes(type).filter((type) => type.isTypeOf === undefined)
+
+    if (
+      resolveTypeImplemented === false &&
+      features.abstractTypes?.resolveType === true &&
+      !features.abstractTypes.isTypeOf
+    ) {
+      console.error(new Error(`${kind} "${type.name}" is missing a \`resolveType\` implementation.`))
+    }
+
+    if (
+      typesWithoutIsTypeOf.length > 0 &&
+      features.abstractTypes?.isTypeOf === true &&
+      !features.abstractTypes.resolveType
+    ) {
+      console.error(
+        new Error(
+          `${kind} "${
+            type.name
+          }" has some members missing an \`isTypeOf\` implementation: ${typesWithoutIsTypeOf
+            .map((t) => `"${t.name}"`)
+            .join(', ')}`
+        )
+      )
+    }
+
+    if (
+      (resolveTypeImplemented === false || typesWithoutIsTypeOf.length > 0) &&
+      features.abstractTypes?.isTypeOf === true &&
+      features.abstractTypes?.resolveType === true
+    ) {
+      console.error(
+        new Error(
+          `${kind} "${
+            type.name
+          }" is either missing a \`resolveType\` implementation or its members are missing an \`isTypeOf\` implementation: ${typesWithoutIsTypeOf
+            .map((t) => `"${t.name}"`)
+            .join(', ')}`
+        )
+      )
+    }
+  })
+}
```

Comment:
Not sure whether these guards shouldn't be throws instead of console.error's. The previous guard on resolveType was a console.error, but I suppose that was mainly to not prevent typegen from running.
As this guard is run _after_ typegen is run, it shouldn't block anything critical.

Is there any reason why we wouldn't wanna throw?

Reply:
Yeah throwing but seeing as this isn't an operational error (its user-land) there's little value IMO in having a stack trace reported. I guess since we're only a library the best we can do is signal this category of error for upstream to better filter on.

So for example `throw new ApplicationError(...)` which would include some structured data like:

```
code: 'NEXUS_ERR_ABSTRACT_TYPES_MISSING_RESOLVE_TYPE'
```

Known error cases shouldn't generally have stack traces presented to users is my thinking.

Reply:
It now throws in production otherwise logs